core/src/main/java/org/wso2/testgrid/core/command/GenerateTestPlanCommand.java

```diff
@@ -628,4 +631,35 @@ private Path getTestGridYamlLocation(String directory) {
             return defaultYamlPath;
         }
     }
+
+    /**
+     * Validate the testgridYaml. It must contain at least one infra provisioner, and one scenario.
+     * @param testgridYaml TestgridYaml object
+     * @return True or False, based on the validity of the testgridYaml
+     */
+    private boolean validateTestgridYaml(TestgridYaml testgridYaml) {
+        Boolean isValidTestgridYaml = true;
```

Comment:
We can optimize this code to directly return true/false instead of using a boolean variable. WDYT?

Reply:
Fixed with bf55b4e